### DISCUSSION TITLE
Can't get eslint to ignore all but certain src/* files

### DISCUSSION THREAD

**Discussion by andrewfritz86** [eslint/eslint#15145] [Q&A] [answered]
Hi - I'd like to have eslint ignore my whole source folder by default, and my team will be adding files that should no longer be ignored as we get to fixing them. I am using eslint `7.32.0`

App structure

```
package.json
src
.eslintignore
```

I call eslint via a script defined in package.json

```
"lint": "eslint src"
```

Then, in .eslintignore, I would like to ignore the entire src folder, EXCEPT `src/components/foo/foo.js`

My .eslintignore

```
src/*
!src/components/foo/foo.js
```

I then run `yarn lint` and receive:

```
You are linting "src", but all of the files matching the glob pattern "src" are ignored.

If you don't want to lint these files, remove the pattern "src" from the list of arguments passed to ESLint.
```

Is this possible? I am trying to follow the example given [here](https://eslint.org/docs/user-guide/configuring/ignoring-code)
```
# Allowlist 'test.js' in the '.build' folder
# But do not allow anything else in the '.build' folder to be linted
!.build
.build/*
!.build/test.js
```

Basically, in place of allowing `test.js` within build, but then ignoring the rest of build, I'd like to allow `src/components/foo/foo.js` but ignore everything else in `src`. I have a feeling this has to do with how I am calling eslint. Thanks!

**Comment by nzakas** [comment, marked as answer]
IIRC, the correct way to do this would be something like:

```
src/**/*.js
!src/components/foo/foo.js
```

I think you can’t omit a file from a directory but you can omit a file from a group of files.

**Reply by andrewfritz86** [reply]
Thank you! That's the fix.
